main.go

```diff
@@ -171,35 +172,21 @@ func main() {
 	// Enqueue initial request
 	updater.EnqueueRequest(&lib.Request{*urlStr})
 
-	reader := bufio.NewReader(os.Stdin)
-	running := true
-	for running {
-		cmd, err := reader.ReadString('\n')
-		if err != nil {
-			fmt.Printf("error reading from stdin: %v\n", err)
-			running = false
-			break
-		}
-
-		// Don't bother processing commands if a worker has failed and
-		// cannot be recovered
-		if workerErr != nil {
-			running = false
-			break
-		}
+	signalChan := make(chan os.Signal, 1)
```

Comment:
Why is this a buffered chan? Is it to prevent multiple routines being spun up when a signal is received?